pkg/test/spacebindingrequest/spacebindingrequest.go

```diff
@@ -0,0 +1,67 @@
+package spacebindingrequest
+
+import (
+	"time"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/gofrs/uuid"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type Option func(spaceRequest *toolchainv1alpha1.SpaceBindingRequest)
+
+func NewSpaceBindingRequest(name, namespace string, options ...Option) *toolchainv1alpha1.SpaceBindingRequest {
+	spaceBindingRequest := &toolchainv1alpha1.SpaceBindingRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			UID:       types.UID(uuid.Must(uuid.NewV4()).String()),
+		},
+	}
+	for _, apply := range options {
+		apply(spaceBindingRequest)
+	}
+	return spaceBindingRequest
+}
+
+func WithMUR(mur string) Option {
+	return func(spaceBindingRequest *toolchainv1alpha1.SpaceBindingRequest) {
+		spaceBindingRequest.Spec.MasterUserRecord = mur
+	}
+}
+
+func WithSpaceRole(spaceRole string) Option {
+	return func(spaceBindingRequest *toolchainv1alpha1.SpaceBindingRequest) {
+		spaceBindingRequest.Spec.SpaceRole = spaceRole
+	}
+}
+
+func WithLabel(key, value string) Option {
+	return func(space *toolchainv1alpha1.SpaceBindingRequest) {
+		if space.ObjectMeta.Labels == nil {
+			space.ObjectMeta.Labels = map[string]string{}
+		}
+		space.ObjectMeta.Labels[key] = value
+	}
+}
+
+func WithDeletionTimestamp() Option {
+	return func(spaceBindingRequest *toolchainv1alpha1.SpaceBindingRequest) {
+		now := metav1.NewTime(time.Now())
+		spaceBindingRequest.DeletionTimestamp = &now
+	}
+}
+
+func WithFinalizer() Option {
+	return func(spaceBindingRequest *toolchainv1alpha1.SpaceBindingRequest) {
+		controllerutil.AddFinalizer(spaceBindingRequest, toolchainv1alpha1.FinalizerName)
+	}
+}
+
+func WithCondition(c toolchainv1alpha1.Condition) Option {
+	return func(spaceBindingRequest *toolchainv1alpha1.SpaceBindingRequest) {
+		spaceBindingRequest.Status.Conditions = append(spaceBindingRequest.Status.Conditions, c)
```

Comment:
you could use `AddOrUpdateStatusConditions` function for that
https://github.com/codeready-toolchain/toolchain-common/blob/c425b7e6bc32a44874fd49d54e4565e724a8368b/pkg/condition/condition.go#L14-L16
but it's just a detail :-)

Reply:
Let's use that. See https://github.com/codeready-toolchain/toolchain-common/pull/351/commits/43de5f98a768e0ee40feb6b26371b77d4b0c22bf

Thanks for the suggestion.